### PR TITLE
Volume host refactor

### LINKED ISSUE #60338: Figure out a way to manage the discrepancy on windows nodes from the linux node
https://github.com/kubernetes/kubernetes/pull/60275/files/2d942dab68b64e684dd2b75232cf9daabc6e0a95#diff-10055ae93a8699af13ceba0482fc43c3

I am expecting we have more windows specific config like code coming to Kubelet and other node components. We should figure out a way to handle this, so that the code can be easily managed? Instead of letting windows-special handling code like this scattered everywhere in our codebase. 

cc/ @feiskyer @michmike  @kubernetes/sig-windows-bugs

### PR DESCRIPTION
**Pull request by raphminkyu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Refactor platform specific code into separate files (*_linux.go / *_windows.go / *_others.go). Referring to #108765

#### Which issue(s) this PR fixes:

Fixes #60338

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by raphminkyu** [comment]
/do-not-merge/work-in-progress

File: pkg/kubelet/volume_host_others.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//go:build !windows
```
Comment (kerthcet) [inline pkg/kubelet/volume_host_others.go:1]: Missed the license.

File: pkg/kubelet/volume_host_windows.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+//go:build windows
```
Comment (kerthcet) [inline pkg/kubelet/volume_host_windows.go:1]: ditto.

File: pkg/kubelet/volume_host.go [hunk 3]
```diff
@@ -129,10 +127,7 @@ func (kvh *kubeletVolumeHost) GetPodsDir() string {
 }
 
 func (kvh *kubeletVolumeHost) GetPodVolumeDir(podUID types.UID, pluginName string, volumeName string) string {
-	dir := kvh.kubelet.getPodVolumeDir(podUID, pluginName, volumeName)
-	if runtime.GOOS == "windows" {
-		dir = util.GetWindowsPath(dir)
-	}
+	dir := getPodHostsPath(kvh, podUID, pluginName, volumeName)
 	return dir
```
Comment (endocrimes) [inline pkg/kubelet/volume_host.go:131]:
```suggestion
	return getPodHostsPath(kvh, podUID, pluginName, volumeName)
```

**Review by endocrimes — CHANGES_REQUESTED** [review]
small change, otherwise LGTM

**Pull request closed without merging**